aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/AwsCustomizableOperationDecorator.kt

```diff
@@ -37,12 +38,8 @@ class CustomizableOperationTestHelpers(runtimeConfig: RuntimeConfig) :
     override fun section(section: CustomizableOperationSection): Writable =
         writable {
             if (section is CustomizableOperationSection.CustomizableOperationImpl) {
-                if (section.operationShape == null) {
+                if (section.runtimeMode == SmithyRuntimeMode.Middleware) {
```

Comment:
Only re-purposed two states out of the quad states, hence I totally see your point. They are being used in the wrong context, hence look confusing. Since it's true or false, a boolean works.

Reply:
Fixed in c9b1a85.